homeassistant/components/tplink/entity.py

```diff
@@ -149,34 +163,11 @@ def __init__(
 
     def _get_unique_id(self) -> str:
         """Return unique ID for the entity."""
-        device = self._device
-        if not isinstance(self, SensorEntity) and self._feature is not None:
-            feature_id = self._feature.id
-            # Special handling for primary state attribute (backwards compat for the main switch).
-            if feature_id == PRIMARY_STATE_ID:
-                return legacy_device_id(device)
-
-            return f"{legacy_device_id(device)}_{feature_id}"
-
-        # Light entities were handled historically differently
-        if isinstance(self, LightEntity):
-            unique_id = device.mac.replace(":", "").upper()
-            # For backwards compat with pyHS100
-            if device.device_type is DeviceType.Dimmer and isinstance(
-                device, IotDevice
-            ):
-                # Dimmers used to use the switch format since
-                # pyHS100 treated them as SmartPlug but the old code
-                # created them as lights
-                # https://github.com/home-assistant/core/blob/2021.9.7/homeassistant/components/tplink/common.py#L86
-                unique_id = legacy_device_id(device)
-            return unique_id
-
-        # For legacy sensors, we construct our IDs from the entity description
-        if isinstance(self, SensorEntity) and self.entity_description is not None:
-            return f"{legacy_device_id(device)}_{self.entity_description.key}"
-
-        return legacy_device_id(device)
+        # The light has it's own implementation and switch and sensor use
+        # the FeatureEntity implementation. When a new non-feature based
+        # platform is added it should provide an implementation here to
+        # return legacy_device_id(self._device)
```

Comment:
```suggestion
        # The light has historically it's own implementation and feature-based entities
        # use feature ids to create unique ids.
        # When a new non-feature based platform is added,
        # it needs to implement this.
```

Reply:
Done in https://github.com/home-assistant/core/pull/120064